jobs.groovy.j2

```diff
@@ -80,688 +83,645 @@
-
-{# create tabs inside the branch folder                                      #}
-{% for name, v  in cfg.views.iteritems()                                    -%}
-  {{ view(name, v.regex, v.description) }}
-{% endfor                                                                    %}
-
-{# ------------------------------------------------------------------------- #}
-{# JOB DEFINITIONS BELOW - THESE WILL BE ADDED TO A MULTIJOB PHASE           #}
-{# --------------------------------------------------------------------------#}
-
-{# iterate over every job_type  #}
-{% for job_type, job_type_values  in cfg.job_type.iteritems()               -%}
-{%  for job_name, job_values  in job_type_values.iteritems()                -%}
-
-{# apply config related to 'run_trial' jobs                                  #}
-{%    if job_type == 'run_trial'                                            -%}
-{%      for _module in job_values.with_modules                              -%}
-{%        set _job_name = job_name + '_' + _module                          -%}
-  job("${dashProject}/${dashBranchName}/{{ _job_name }}") {
-    parameters {
-{# we pass the 'MODULE' parameter as the flocker module to test with trial   #}
-        textParam("MODULE", "{{ _module }}", "Module to test" )
-        textParam("TRIGGERED_BRANCH", "${branchName}", "Branch that triggered this job" )
+// create tabs inside the branch folder
+def build_tabs(dashBranchName) {
+    for (entry in GLOBAL_CONFIG.views) {
+        name = "${dashProject}/${dashBranchName}/${entry.key}"
+        println("creating tab ${name}...")
+        listView(name) {
+            description(entry.value.description)
+            filterBuildQueue()
+            filterExecutors()
+            jobs {
+                regex(entry.value.regex)
+            }
+            columns {
+                status()
+                weather()
+                name()
+                lastSuccess()
+                lastFailure()
+                lastDuration()
+                buildButton()
+            }
+        }
+    }
+}
+
+
+def define_job(dashBranchName, branchName, job_type, job_name, job_values) {
+    // apply config related to 'run_trial' jobs
+    if (job_type == 'run_trial') {
+        for (_module in job_values.with_modules) {
+            _job_name = job_name + '_' + _module
+            job("${dashProject}/${dashBranchName}/${_job_name}") {
+                parameters {
+                    // we pass the 'MODULE' parameter as the flocker module to test with trial
+                    textParam("MODULE", _module, "Module to test" )
+                    textParam("TRIGGERED_BRANCH", branchName,
+                              "Branch that triggered this job" )
+                }
+                // limit execution to jenkins slaves with a particular label
+                label(job_values.on_nodes_with_labels)
+                directories_to_delete = ["${WORKSPACE}/_trial_temp",
```

Comment:
These are doing interpolation, which they shouldn't be. There are a number of occurrences.